File: src/graphics/movable_image/movable_image.cpp
#include "movable_image.h"
#include <QPainter>

const int rectSize = 100;

MovableImage::MovableImage(QObject *parent) : QObject(parent), QGraphicsItem() {

  this->setFlag(QGraphicsItem::ItemIsMovable);
  this->setFlag(QGraphicsItem::ItemIsSelectable);
  this->setFlag(QGraphicsItem::ItemSendsScenePositionChanges);
}

QRectF MovableImage::boundingRect() const {
  return QRectF(-rectSize / 2, -rectSize / 2, rectSize, rectSize);
}

void MovableImage::paint(QPainter *painter,
                         const QStyleOptionGraphicsItem *option,
                         QWidget *widget) {
  Q_UNUSED(option);
  Q_UNUSED(widget);

  auto brushColor = isSelected() ? Qt::darkBlue : Qt::darkGray;
  painter->setPen(Qt::black);
  painter->setBrush(brushColor);
  painter->drawRect(-rectSize / 2, -rectSize / 2, rectSize, rectSize);
}

File: src/graphics/python/movable_image/movable_image.py
from PyQt6 import QtGui
from PyQt6.QtCore import QRectF, QObject, Qt
from PyQt6.QtGui import QPen, QColor
from PyQt6.QtWidgets import QGraphicsItem

rect_size = 100

class MovableImage(
    QGraphicsItem, QObject):
    def __init__(self, parent=None):
        super(QGraphicsItem, self).__init__()
        #super(QObject, self).__init__(parent)
        self.setFlag(QGraphicsItem.GraphicsItemFlag.ItemIsMovable)
        self.setFlag(QGraphicsItem.GraphicsItemFlag.ItemIsSelectable)
        self.setFlag(QGraphicsItem.GraphicsItemFlag.ItemSendsScenePositionChanges)
    
    def boundingRect(self):
        return QRectF(-rect_size / 2, -rect_size / 2, rect_size, rect_size)

    def paint(self, painter, option, widget):
        brush_color = self.isSelected() and QtGui.QColorConstants.DarkGreen or QtGui.QColorConstants.DarkGray
        painter.setPen(QPen(QColor("black")))
        painter.setBrush(brush_color)
        painter.drawRect(-rect_size / 2, -rect_size / 2, rect_size, rect_size)
